### PR TITLE
Add optimization section to documentation

### PR DESCRIPTION
**Pull request by deadeyegoodwin**

### CODE REVIEW

File: docs/optimization.rst [hunk 1]
```diff
@@ -0,0 +1,276 @@
...
+  Concurrency: 2, 204.8 infer/sec, latency 9807 usec
+  Concurrency: 3, 204.2 infer/sec, latency 14846 usec
+  Concurrency: 4, 199.6 infer/sec, latency 20499 usec
+
+The results show that our non-optimized model configuration gives a
+throughput of about 200 inferences per second. Note how there is a
+significant throughput increase going from one concurrent request to
+two concurrent requests and then throughput levels off. With one
+concurrent request the inference server is idle during the time when
+the response is returned to the client and the next response is
```
Comment (tanmayv25) [inline docs/optimization.rst]: and the next request

File: docs/optimization.rst [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+The TensorRT optimization provided 2x throughput inprovement while
+cutting latency in half. The benefit provided by TensorRT will vary
+based on the model, but in general it can provide significant
+performance improvement.
+
+TensorFlow with TensorRT Optimization
+.....................................
+
+TensorRT optimization applied to a TensorFlow model works similarly to
+TensorRT and ONNX described above. To enable TensorRT optimization you set the model configuration appropriately. For TensorRT optimization of TensorFlow models that are several options that you can enable, including selection the precision. For example::
```
Comment (tanmayv25) [inline docs/optimization.rst]: To enable TensorRT optimization you must set the model configuration appropriately.

File: docs/perf_client.rst [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+  # EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+  # PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR
+  # PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY
+  # OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  # (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+  # OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+.. _section-perf-client:
+
+perf\_client
```
Comment (tanmayv25) [inline docs/perf_client.rst]: This looks like a place where we must document our observations with synchronous and asynchronous modes. Once this [PR](https://github.com/NVIDIA/tensorrt-inference-server/pull/822) is merged.

Comment (deadeyegoodwin) [reply docs/perf_client.rst]: Agreed and I will file a follow up ticket for that additional documentation. Before we do that I want to add some nightly performance testing for perf_client that compares it to simple_perf_client.

File: docs/optimization.rst [hunk 4]
```diff
@@ -0,0 +1,279 @@
...
+fully utilizing the GPU and so adding additional model instances does
+not provide any performance advantage. In general the benefit of the
+dynamic batcher and multiple instances is model specific, so you
+should experiment with perf\_client to determine the settings that
+best satisfy your throughput and latency requirements.
+
+Framework-Specific Optimization
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The inference server has several optimization setting that apply only
```
Comment (tanmayv25) [inline docs/optimization.rst]: settings

File: docs/optimization.rst [hunk 5]
```diff
@@ -0,0 +1,279 @@
...
+performance improvement.
+
+TensorFlow with TensorRT Optimization
+.....................................
+
+TensorRT optimization applied to a TensorFlow model works similarly to
+TensorRT and ONNX described above. To enable TensorRT optimization you
+must set the model configuration appropriately. For TensorRT
+optimization of TensorFlow models that are several options that you
+can enable, including selection the precision. For example::
```
Comment (tanmayv25) [inline docs/optimization.rst]: including the precision selection OR including the selection of the precision

**Pull request merged**